src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -563,19 +563,47 @@ public sealed override ImmutableArray<TypeWithAnnotations> TupleElementTypesWith
         public sealed override ImmutableArray<FieldSymbol> TupleElements
             => IsTupleType ? TupleData!.TupleElements(this) : default;
 
+        /// <summary>
+        /// For tuple fields that aren't TupleElementFieldSymbol or TupleErrorFieldSymbol, we cache their tuple element index.
+        /// This supports <see cref="FieldSymbol.TupleElementIndex"/>.
+        /// For those fields, we map from their definition to an index.
+        /// </summary>
+        public SmallDictionary<FieldSymbol, int>? TupleFieldDefinitionsToIndexMap
+        {
+            get
+            {
+                if (!IsTupleType)
+                {
+                    return null;
+                }
+
+                if (!IsDefinition)
+                {
+                    return this.OriginalDefinition.TupleFieldDefinitionsToIndexMap;
+                }
+
+                return TupleData!.GetFieldDefinitionsToIndexMap(this);
+            }
+        }
+
         public TMember? GetTupleMemberSymbolForUnderlyingMember<TMember>(TMember underlyingMemberOpt) where TMember : Symbol
         {
             return IsTupleType ? TupleData!.GetTupleMemberSymbolForUnderlyingMember(underlyingMemberOpt) : null;
         }
```

Comment:
>SymbolEqualityComparer.ConsiderEverything [](start = 98, length = 41)

Can we use ```ReferenceEqualityComparer``` here?

Reply:
Did you re-activate this comment?
I may be confused because it appears in a weird place in CodeFlow, but I thought I've addressed it already with change on line 616 to `new SmallDictionary<FieldSymbol, int>(ReferenceEqualityComparer.Instance)`

---
In reply to: [582411270](https://github.com/dotnet/roslyn/pull/44231#discussion_r582411270) [](ancestors = 582411270)

Reply:
> Did you re-activate this comment?

I do not think I did. Closing.

---
In reply to: [583835798](https://github.com/dotnet/roslyn/pull/44231#discussion_r583835798) [](ancestors = 583835798,582411270)